src/app-layer-parser.c

```diff
@@ -655,7 +662,7 @@ uint64_t AppLayerTransactionGetActiveLogOnly(Flow *f, uint8_t flags)
     AppLayerParserProtoCtx *p = &alp_ctx.ctxs[f->protomap][f->alproto];
 
     if (p->logger == TRUE) {
```

Comment:
On 17-12-15 17:46, Jason Ish wrote:
> Currently yes. They will be updated as each direction reaches
> completion, whether or not anything was logged (as thats up to the
> logging callback, but the transition is based on readyness). Then we
> move forward to the next one. So you are getting the same behaviour as
> before which waits for both to become ready before anything is logged.
> 
> So it should behave exactly the same as now. Unless I'm missing some
> special case on the current setup?

The id tracking is complex is master already, but gets really hard to
follow now. It's not very clear to me when/how the 2 log id's are
updated now. I would like to see a cleaner implementation of this, esp
in the main logging loop.